### PR TITLE
Add __eq__ method to PipelineFile to allow sane comparison of instanc…

### PR DESCRIPTION
**Pull request by lwgordonimos**
…es. Add __bool__ to PipelineFileCollection to behave as expected when evaluating truthiness.

~~~DO NOT MERGE JUST YET~~~ ready for review

### CODE REVIEW

File: test_aodncore/pipeline/test_files.py [hunk 1]
```diff
@@ -11,33 +11,36 @@
...
         # Test compliance checking
         check_runner = get_child_check_runner(PipelineFileCheckType.NC_COMPLIANCE_CHECK, None, self.mock_logger,
                                               {'checks': ['cf']})
         check_runner.run(PipelineFileCollection(self.pipelinefile))
         assertCountEqual(self, dict(self.pipelinefile.check_result).keys(), ['compliant', 'errors', 'log'])
 
+    def test_equal_files(self):
+        duplicate_file = PipelineFile(GOOD_NC)
+        self.assertFalse(id(self.pipelinefile) == id(duplicate_file))
+        self.assertTrue(self.pipelinefile == duplicate_file)
```
[the remark this answers is not in the document]
Comment (mhidas) [reply test_aodncore/pipeline/test_files.py]: :man_shrugging: (there's no icon for "whatever")

[the remark this answers is not in the document]
Comment (lwgordonimos) [reply test_aodncore/pipeline/test_files.py]: There was no icon for "face palm" either when I found it was PyCharm monkey patching the core unittest module.

**Comment by mhidas** [comment]
@lwgordonimos can you push up that other change (accumulating results in `PipelineFileCollection.update()`)? Then I think we can merge this thing.

**Comment by lwgordonimos** [comment]
@mhidas updated the result, and added fix for bug where it was retaining the original object rather than overwriting (as expected when you've explicitly set "overwrite" to True!).

**Pull request merged**